### ISSUE TITLE
Table Detection

### ISSUE THREAD

**Issue by lzneu** [zylo117/Yet-Another-EfficientDet-Pytorch#16]
感谢作者优秀的工作！
我在使用该实现做table detection表格检测，第66个epoch val_loss降低至0.23，但是预测时并不能取得很好的效果，作者有什么好的建议吗，感谢！

**Comment by zylo117** [comment]
can you run coco_eval.py and share the result?

**Comment by zylo117** [comment]
btw, loss being low doesn't mean the training is good. It could be the anchors are hard to match your gt boxes, so lots of gt boxes have been filtered, then there is no loss at all.

**Comment by lzneu** [comment]
> can you run coco_eval.py and share the result?
> how did you train the model? head_only or all layers?
> 
> If the result is bad, it's very possible that ratio of your groundtruth boxes greatly differs from the anchors ratio.

I've already run coco_eval.py  and set iou_shresh=0.5, the ap is so low..
![企业微信截图_c20228cc-a955-4297-894e-78dea2f8f085](https://user-images.githubusercontent.com/29429803/78974883-8d314e00-7b45-11ea-88e7-92662ef20112.png)
and I trained the model  all layers with no resume weights

**Comment by zylo117** [comment]
> > can you run coco_eval.py and share the result?
> > how did you train the model? head_only or all layers?
> > If the result is bad, it's very possible that ratio of your groundtruth boxes greatly differs from the anchors ratio.
> 
> I've already run coco_eval.py and set iou_shresh=0.5, the ap is so low..
> ![企业微信截图_c20228cc-a955-4297-894e-78dea2f8f085](https://user-images.githubusercontent.com/29429803/78974883-8d314e00-7b45-11ea-88e7-92662ef20112.png)
> and I trained the model all layers with no resume weights

Then that's what I mentioned earlier, the loss is fake, because you used a inappropriate anchors.

Please pull the latest code and change anchor scales and ratios following the instruction.

**Comment by lzneu** [comment]
> > > can you run coco_eval.py and share the result?
> > > how did you train the model? head_only or all layers?
> > > If the result is bad, it's very possible that ratio of your groundtruth boxes greatly differs from the anchors ratio.
> > 
> > 
> > I've already run coco_eval.py and set iou_shresh=0.5, the ap is so low..
> > ![企业微信截图_c20228cc-a955-4297-894e-78dea2f8f085](https://user-images.githubusercontent.com/29429803/78974883-8d314e00-7b45-11ea-88e7-92662ef20112.png)
> > and I trained the model all layers with no resume weights
> 
> Then that's what I mentioned earlier, the loss is fake, because you used a inappropriate anchors.
> 
> Please pull the latest code and change anchor scales and ratios following the instruction.

I'll do this. Thank you!

**Comment by firefox1031** [comment]
@lzneu 我在自己的数据集上也遇到了同样的问题，loss很低但是AP也很低，几乎为0。请问你的问题解决了么？是否是anchor设置的问题导致的？

**Comment by lzneu** [comment]
> @lzneu 我在自己的数据集上也遇到了同样的问题，loss很低但是AP也很低，几乎为0。请问你的问题解决了么？是否是anchor设置的问题导致的？

Not yet. I turn to segmention model.

**Comment by zylo117** [comment]
There was a unfixed bug when you created this issue, but it's fixed by now.

Segmentation models usually perform worse than detection model at bbox regression

**Comment by lzneu** [comment]
> Segmentation models usually perform worse than detection model at bbox regression

That‘s Great! I'll try it this weekend.

**Comment by lzneu** [comment]
> There was a unfixed bug when you created this issue, but it's fixed by now.
> 
> Segmentation models usually perform worse than detection model at bbox regression

When I pulled the lastest code and changed the anchor params, I've trained a custom model, the AP0.5=0.782 in my own dataset. Your work is great!
